src/applications/edu-benefits/components/AutosuggestField.jsx

```diff
@@ -0,0 +1,296 @@
+}
+
+export default class AutosuggestField extends React.Component {
+  constructor(props) {
+    super(props);
+    const { uiSchema, schema, formData } = props;
+    const input = getInput(formData, uiSchema, schema);
+    const uiOptions = uiSchema['ui:options'];
+
+    let options = [];
+    let suggestions = [];
+
+    if (!uiOptions.getOptions) {
+      this.useEnum = true;
+      options = schema.enum.map((id, index) => {
+        return {
+          id,
+          label: uiOptions.labels[id] || schema.enumNames[index]
+        };
+      });
+      suggestions = this.getSuggestions(options, input);
+    }
+
+    const debounceRate = uiOptions.debounceRate === undefined ? 1000 : uiOptions.debounceRate;
+    this.debouncedGetOptions = debounce(debounceRate, this.getOptions);
+
+    this.state = {
+      listLabel: uiOptions.listLabel,
+      loading: false,
+      options,
+      input,
+      suggestions
+    };
+  }
+
+  componentDidMount() {
+    if (!this.props.formContext.reviewMode) {
+      this.getOptions();
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmounted = true;
+    this.debouncedGetOptions.cancel();
+  }
+
+  getOptions = (inputValue) => {
+    const getOptions = this.props.uiSchema['ui:options'].getOptions;
+    if (getOptions) {
+      getOptions(inputValue)
+        .then(({ options, searchTerm }) =>
+          // match response to current request
+          searchTerm === this.state.searchTerm &&
+          this.setOptions(options));
+    }
+  }
+
+  setOptions = (options) => {
+    if (!this.unmounted) {
+      const optionsOrDefault = options.length > 0 ?
+        options :
+        [{ value: null, label: 'No results' }];
+      this.setState({
+        loading: false,
+        searchTerm: '',
+        options: optionsOrDefault,
+        suggestions: this.getSuggestions(optionsOrDefault, this.state.input)
+      });
+    }
+  }
+
+  getSuggestions = (options, value) => {
+    if (value) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      const sortedOptions = sortListByFuzzyMatch(value, options).slice(0, uiOptions.maxOptions);
+      return sortedOptions;
+    }
+
+    return options;
+  }
+
+  getFormData = (suggestion) => {
+    if (this.useEnum) {
+      return suggestion.id;
+    }
+
+    // When freeInput is true, we'll return the label to the api instead of the id
+    if (this.props.uiSchema['ui:options'].freeInput) {
+      return suggestion.label;
+    }
+
+    return _.set('widget', 'autosuggest', suggestion);
+  }
+
+  handleClearSelection = (e, downshiftClear) => {
+    e.preventDefault();
+    // using Downshift's clear sets the correct focus state after using clear
+    downshiftClear(() => this.setState({
+      input: '',
+      searchTerm: '',
+      loading: false
+    }));
+  }
+
+  handleInputValueChange = (inputValue) => {
+    if (inputValue !== this.state.input) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      if (uiOptions.queryForResults) {
```

Comment:
One significant behavior change: this queries on every input change. The previous version would fetch the data once and then do a fuzzy find on it. Not sure how we'll reconcile this if we want to pass this as an update into the forms library.